roqoqo-iqm/tests/integration/backend.rs

```diff
@@ -33,6 +33,30 @@ fn init_backend() {
     }
 }
 
+#[test]
+fn test_register_initialization() {
+    if env::var("IQM_TOKENS_FILE").is_ok() {
+        let device = DemoDevice::new();
+        let backend = Backend::new(device.into(), None).unwrap();
+
+        let mut qc = Circuit::new();
+        qc += ControlledPauliZ::new(0, 2);
+        qc += DefinitionBit::new("my_reg1".to_string(), 5, true);
+        qc += DefinitionBit::new("my_reg2".to_string(), 7, true);
+        qc += MeasureQubit::new(2, "my_reg1".to_string(), 2);
+
+        let (bit_registers, _float_registers, _complex_registers) =
+            backend.run_circuit(&qc).unwrap();
+
+        let reg2_result = bit_registers.get("my_reg2").unwrap();
+        let expected_output = vec![vec![false; 7]];
+
+        assert_eq!(*reg2_result, expected_output);
+    } else {
+        panic!("no IQM_TOKENS_FILE env var found.")
```

Comment:
I would suggest printing a warning and passing, so that the pipeline still passes